frontend/components/pages/CourseDetail.tsx

```diff
@@ -41,23 +37,32 @@ export type HandleChangeVideo = (
   videoIndex: number,
 ) => void
 
+export type CreateQuestionErrorType = { title: string; content: string }
+
 export function CourseDetail({
   courseData,
   session,
+  questions,
 }: {
-  courseData: CourseDetailPropsType
+  courseData: CourseWithSectionsType
   session: Session | null
+  questions?: QuestionType[]
 }) {
+  const router = useRouter()
   const { showErrorToast } = useCustomToast()
   const userId = session?.user?.id
+  const searchParams = useSearchParams()
+  const searchedVideoId = searchParams.get('videoId')
 
   const [isWatched, setIsWatched] = useState<{ [key: string]: boolean }>({})
   const [isChecked, setIsChecked] = useState<{ [key: string]: boolean }>({})
   const [isLoading, setIsLoading] = useState(false)
   const [isFavorited, setIsFavorited] = useState(false)
-  const [questions, setQuestions] = useState<QuestionType[]>()
+  const [createQuestionErrors, setCreateQuestionErrors] =
+    useState<CreateQuestionErrorType>({ title: '', content: '' })
+  const [questionPage, setQuestionPage] = useState('QuestionList')
```

Comment:
どういうパターンがあるか定義してもらっていいですか？

```suggestion
  const [questionPage, setQuestionPage] = useState<QuestionPageType>('QuestionList')
  
  type QuestionPageType = 'questionList' | '他の名前' | '他にもあれば'
  
```

これunion型って言うんで覚えておいてください！
https://typescriptbook.jp/reference/values-types-variables/union


今の実装だとどんな文字列でも入ってきてしまうので😎

Reply:
でどっかtypesのフォルダの中に切り出しお願いします！

Reply:
@issiyda 
union型で定義して、types/QuestionTypeで定義しています！